src/cpp/SearchCriteria.h

```diff
@@ -0,0 +1,189 @@
+//
+// Created by Dmytro Klishch on 10/7/22.
+//
+
+#ifndef KNOWYOURKNOWLEDGEMYFORK_SEARCHCRITERIA_H
+#define KNOWYOURKNOWLEDGEMYFORK_SEARCHCRITERIA_H
+
+#include "doctest.h"
+
+#include <map>
+#include <vector>
+#include <string>
+
+template<typename... T>
+using searchCondition = std::function<bool(std::variant<T...>)>;
```

Comment:
No include for headers that are required for std::function and std::variant
Please check e.g. cppreference and always include the headers needed for all used types/functions/...

Reply:
Hmm, it worked for me without directly including them. 
Just curious, do we have this difference because of differences in compilers or has something else caused this?

Reply:
Some headers (including those from standard C++ library) may include other headers as implementation detail. This means that sometimes certain types/functions/... work even without proper includes - because they were included in another header. However you should not count on this - because it may not work in another compiler, or even in future versions of the same compiler. Also proper includes can serve as documentation - you can see which capabilities are used in each file.